tools/crm_simulate.c

```diff
@@ -790,207 +831,103 @@ profile_all(const char *dir, long long repeat, pe_working_set_t *data_set, char
-
-                options.modified++;
-                options.quorum = strdup(optarg);
-                break;
-            case 'g':
-                options.modified++;
-                options.ticket_grant = g_list_append(options.ticket_grant, optarg);
-                break;
-            case 'r':
-                options.modified++;
-                options.ticket_revoke = g_list_append(options.ticket_revoke, optarg);
-                break;
-            case 'b':
-                options.modified++;
-                options.ticket_standby = g_list_append(options.ticket_standby, optarg);
-                break;
-            case 'e':
-                options.modified++;
-                options.ticket_activate = g_list_append(options.ticket_activate, optarg);
-                break;
-            case 'a':
-                options.all_actions = TRUE;
-                break;
-            case 's':
-                options.process = TRUE;
-                show_scores = TRUE;
-                break;
-            case 'U':
-                options.process = TRUE;
-                show_utilization = TRUE;
-                break;
-            case 'j':
-                options.print_pending = TRUE;
-                break;
-            case 'S':
-                options.process = TRUE;
-                options.simulate = TRUE;
-                break;
-            case 'X':
-                options.store = TRUE;
-                options.process = TRUE;
-                options.simulate = TRUE;
-                break;
-            case 'R':
-                options.process = TRUE;
-                break;
-            case 'D':
-                if (options.dot_file) {
-                    free(options.dot_file);
-                }
-
-                options.process = TRUE;
-                options.dot_file = strdup(optarg);
-                break;
-            case 'G':
-                if (options.graph_file) {
-                    free(options.graph_file);
-                }
-
-                options.process = TRUE;
-                options.graph_file = strdup(optarg);
-                break;
-            case 'I':
-                options.input_file = optarg;
-                break;
-            case 'O':
-                options.output_file = optarg;
-                break;
-            case 'P':
-                options.test_dir = optarg;
-                break;
-            case 'N':
-                repeat_s = optarg;
-                break;
-            default:
-                ++argerr;
-                break;
-        }
+    for (int i = 0; i < args->verbosity; i++) {
+        crm_bump_log_level(argc, argv);
+    }
+
+    if (args->version) {
+        /* FIXME:  When crm_simulate is converted to use formatted output, this can go. */
+        pcmk__cli_help('v', CRM_EX_USAGE);
+        goto done;
     }
 
     if (optind > argc) {
-        ++argerr;
+        fprintf(stderr, "%s", g_option_context_get_help(context, TRUE, NULL));
```

Comment:
needs to be freed :)

Reply:
Heh, I even just got finished patching this elsewhere.